### PR TITLE
Add counsel-up-level command

### PR DESCRIPTION
**Pull request by czipperz**
This is the equivalent of helm's `C-l`.

### CODE REVIEW

**Comment by abo-abo** [comment]
How is this different from `counsel-up-directory` (bound to `C-<backspace>`)?

**Comment by czipperz** [comment]
This is similar to `ivy-backward-kill-word` when backward kill word is called at the end of the prompt line.

**Comment by abo-abo** [comment]
I think I like `counsel-up-level` behavior more. Could you please rework your change as a custom variable that makes `counsel-up-directory` behave as `counsel-up-level`? Make the variable on by default, and when it's off, the behavior of `counsel-up-directory` will be the same as now.

Also, I don't see your email in the Emacs Copyright Assignment list. The CA is needed for changes over 15 lines. See README for more info.

**Comment by czipperz** [comment]
I have already filled out a copyright assignment previously.  I could send you the pdf of it, already signed by John Hsieh on January 18, 2017

**Comment by abo-abo** [comment]
I don't see it yet on `copyright.list`. Could you please ask the copyright clerk to make sure you're there? Also configure `git` so that at least the commit email matches the on the CA.

File: counsel.el [hunk 1]
```diff
@@ -1588,14 +1588,29 @@ When INITIAL-INPUT is non-nil, use it in the minibuffer during completion."
             :keymap counsel-find-file-map
             :caller 'counsel-find-file))
 
+(defvar counsel-up-directory-level t
```
Comment (kljohann) [inline counsel.el]: Could you make this a `defcustom` with `:type 'boolean` and `:group 'ivy`?

**Comment by CeleritasCelery** [comment]
@czipperz have you managed to get copyright assignment yet? I would love to see this merged.

**Comment by czipperz** [comment]
Ok I'm on the list now.

**Comment by czipperz** [comment]
Okay I rebased it.

**Comment by czipperz** [comment]
Locally (emacs 26.2) test `counsel-find-file-with-dotfiles` fails at bottom of ivy-test.el.  Does anyone know why the first test expects `tests/find-file/dotfiles/foo/`?  It doesn't make much sense to me.

**Comment by czipperz** [comment]
(it fails on abo-abo/master as well as czipperz/master)

**Comment by CeleritasCelery** [comment]
looks like that has been removed from CI for now. LGTM

**Comment by abo-abo** [comment]
Merged, thanks.

**Pull request closed without merging**